x/wasm/ioutils/ioutil.go

```diff
@@ -6,14 +6,14 @@ import (
 	"io"
 
 	errorsmod "cosmossdk.io/errors"
-
-	"github.com/CosmWasm/wasmd/x/wasm/types"
 )
 
+var errLimit = errorsmod.Register("wasm", 29, "exceeds limit")
```

Comment:
let's not register a new error type outside of types/errors . You can just make this a regular stdlib error. In the method call you can check for the concrete type if needed and use a registered type.  Please make sure that this error type is public